lib/os/multi_heap.c

```diff
@@ -49,9 +51,10 @@ void *sys_multi_heap_aligned_alloc(struct sys_multi_heap *mheap,
 	return mheap->choice(mheap, cfg, align, bytes);
 }
 
-void sys_multi_heap_free(struct sys_multi_heap *mheap, void *block)
+int sys_multi_heap_get_heap(struct sys_multi_heap *mheap, void *addr,
```

Comment:
`const struct sys_multi_heap *mheap` ?

Reply:
const struct is  good as well, but in case the heap has not been found NULL will be returned. The higher layer need to check it - and it's easier to overlook it than checking if the exit code is != 0 
can easily change it, its up to you and @andyross ?